language/cc/resolve.go

```diff
@@ -113,24 +122,23 @@ func (*ccLanguage) Imports(c *config.Config, r *rule.Rule, f *rule.File) []resol
 	return imports
 }
 
-// transformIncludePath converts a path to a header file into a string by which the
-// header file may be included, accounting for the library's
-// strip_include_prefix and include_prefix attributes.
+// transformIncludePath converts a path to a header file into a string by which the header file may be included,
+// accounting for the library's strip_include_prefix and include_prefix attributes.
+//
+// libRel is the slash-separated, repo-root-relative path to the directory containing the target.
 //
-// libRel is the slash-separated, repo-root-relative path to the directory
-// containing the target.
+// stripIncludePrefix is the value of the target's strip_include_prefix attribute. If it's "", this has no effect. If
+// it's a relative path (including "."), both libRel and stripIncludePrefix are stripped from rel. If it's an absolute
+// path, the leading '/' is removed, and only stripIncludePrefix is removed from hdrRel.
 //
-// stripIncludePrefix is the value of the target's strip_include_prefix
-// attribute. If it's "", this has no effect. If it's a relative path (including
-// "."), both libRel and stripIncludePrefix are stripped from rel. If it's an
-// absolute path, the leading '/' is removed, and only stripIncludePrefix is
-// removed from hdrRel.
+// includePrefix is the value of the target's include_prefix attribute. It's prepended to hdrRel after
+// stripIncludePrefix is applied.
 //
-// includePrefix is the value of the target's include_prefix attribute.
-// It's prepended to hdrRel after stripIncludePrefix is applied.
+// SPECIAL CASE: regardless of the official Bazel docs, when includePrefix is set and stripIncludePrefix is not set,
+// we treat it as if stripIncludePrefix were set to the library's repo-root-relative path (libRel). This matches
+// observed Bazel behavior.
```

Comment:
The `SPECIAL CASE: regardless of the official Bazel docs` here makes it sound like we're contradicting what the Bazel documentation says, but it doesn't really say what happens. Let's just remove that phrase.